library/oneview_server_hardware_facts.py

```diff
@@ -209,6 +219,7 @@ class ServerHardwareFactsModule(OneViewModule):
     def __init__(self):
         argument_spec = dict(
             name=dict(required=False, type='str'),
+            uri=dict(required=False, type='str'),
```

Comment:
where are you using this uri inside the code..?

Reply:
you will get the server_hardware details particular to URI given
I have updated the library file.
Thanks for pointing out !